### DISCUSSION TITLE
How do we want to read in configuration settings?

### DISCUSSION THREAD

**Discussion by 2320sharon** [SatelliteShorelines/CoastSeg#42] [Q&A]
# How do we want to read in configuration settings?
CoastSeg has a lot of settings that are currently hardcoded, which while it makes it easier for the user to edit them it also has the drawback of cluttering the code. It has been brought up that we want to read the configuration settings from a file.
1. Do we want this file to a `JSON` file or an `ini` file?
- `JSON` files can be easily edited by the user with or without code, but can be tricky when it comes to spacing out items when editing the file directly.
- `ini` files can be manipulated with the ConfigParser module and according to my research are easy to edit both with code and by file. HoloDoodler also uses an `ini` file to read in classes and corresponding colors for drawing.

2. Do we want all the settings in one file or multiple files?
- My recommendation is multiple files because it makes it easier for the user to edit only the things they are about and makes it easier to track what's being manipulated in the code.

**Comment by ddbuscombe** [comment]
Good questions. Definitely would like to discuss tomorrow. Initially, I'm inclined towards ini files for the apparent convenience, and multiple files seems like the best path forward
